pkg/infrastructure/services/dashboards.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	grafanav1 "github.com/integr8ly/grafana-operator/v3/pkg/apis/integreatly/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// GrafanaDashboard is a structure that contains the fetched dashboards
+type GrafanaDashboard struct {
+	Name             string
+	RawJSONDashboard string
+}
+
+const (
+	// dashboardPath which the dashboards are fetched
+	dashboardsPath = "/monitoring/dashboards/"
+
+	// GrafanaDashboardAppName label app name to be used when a GrafanaDashboard is created
+	GrafanaDashboardAppName = "grafana"
+)
+
+// FetchGrafanaDashboards fetches the grafana dashboards from the KogitoService
+func FetchGrafanaDashboards(cli *client.Client, instance v1alpha1.KogitoService) ([]GrafanaDashboard, error) {
```

Comment:
Quick question: Why the `Fetch*` services are exported? Can't we keep then private?

```suggestion
func fetchGrafanaDashboards(cli *client.Client, instance v1alpha1.KogitoService) ([]GrafanaDashboard, error) {
```

Reply:
Hi @ricardozanini , sure. I've changed accordingly